shim/linux/src/entry.c

```diff
@@ -843,10 +845,23 @@ dispatch_vcpu_kvm_get_cpuid2(struct kvm_cpuid2 *const ioctl_args)
 }
 
 static long
-dispatch_vcpu_kvm_get_fpu(struct kvm_fpu *const ioctl_args)
+dispatch_vcpu_kvm_get_fpu(
+    struct shim_vcpu_t const *const vcpu, struct kvm_fpu *const ioctl_args)
 {
-    (void)ioctl_args;
-    return -EINVAL;
+    struct kvm_fpu mut_args;
+    uint64_t const size = sizeof(mut_args);
+
+    if (handle_vcpu_kvm_get_fpu(vcpu, &mut_args)) {
```

Comment:
We need a copy_from_user here to populate mut_args from ioctl_args right? Or is no information passed into the ioctl from userspace?

Reply:
No Information is passed into ioctl, we need to return the information. so handle_vcpu_kvm_get_fpu returns the mut_args with the information and we use platform_copy_to_user function to return from ioctl.